### ISSUE TITLE
Improve (API) documentation

### ISSUE THREAD

**Issue by dexhunter** [hyperledger/fabric-sdk-py#40]
Hi all, I am trying to use Sphinx `autodoc` function/extension, so I need to improve the current documentation of `hfc`. I will update the information on this issue later.

**Comment by viragumathe5** [comment]
I really want to contribute shall I start with it on documentation??

**Comment by dexhunter** [comment]
@viragumathe5 yes! Please have a look

**Comment by viragumathe5** [comment]
Hi, I have made a PR to improve the documentation. @dexhunter would you please review it, please  
https://github.com/hyperledger/fabric-sdk-py/pull/50
Thank You

**Comment by dexhunter** [comment]
Let me give an example,

```python
    async def query_installed_chaincodes(self, requestor, peers,
                                         transient_map=None, decode=True):
        """
        Queries installed chaincode, returns all chaincodes installed on a peer
        :param requestor: User role who issue the request
        :param peers: Names or Instance of the peers to query
        :param transient_map: transient map
        :param decode: Decode the response payload
        :return: A `ChaincodeQueryResponse` or `ProposalResponse`
        """
```

 so at [client.py](https://github.com/hyperledger/fabric-sdk-py/blob/master/hfc/fabric/client.py#L2269-L2279) functions are documented in a Sphinx sylte,

```python
    def add_peer(self, peer):
        """Add peer endpoint to a chain object.
        Args:
             peer: an instance of the Peer class
        """
```

While at [channel.py]() some functions are documented in a different style (Google Python Docstring).

So we need to make our documentation style coherent (all convert to Sphinx style) and add more explanations if possible. Thanks.

[A reference for different styles please click this](https://www.datacamp.com/community/tutorials/docstrings-python)

**Comment by viragumathe5** [comment]
@dexhunter as you asked me to do the link  based auto-documentation for that we need to increase the number of files like I was thinking the documentation 
- Pre-requisites
- Creating your virtual environment 

these are the links that will direct the user to other pages for that we need more files so will it be okay???

From separate files, we can also introduce more auto-generation of documentation ???

**Comment by dexhunter** [comment]
@viragumathe5 Hi, sorry but I don't really understand your question. I am not asking you to "do the link", I am asking help to improve current API documentation. Since we already have tons of APIs/functionalities, I think it will help potential mentees to understand py-sdk better by reading the source code and add some explanations. The links will be auto-generated by Sphinx framework so there is no need to manually add them from my perspective.

**Comment by viragumathe5** [comment]
Yes yes, I am on but at the same time, you had mentioned something like "I want to clean up the documentation page for the easier navigation" on my PR so I am talking about it. I am about to complete the change so I will show the changes in my PR just give me a few minutes

**Comment by viragumathe5** [comment]
Hello, @dexhunter I am facing a problem while adding the documentation like where should I add it in documentation like by making some title for API documentation or just add it in tutorials or should I add examples for each and then write documentation ????
Would you please help me

**Comment by viragumathe5** [comment]
Like take an example for **crypto and vision** should I make a separate API documentation page or add it there straight or make some example and add docs ???

**Comment by dexhunter** [comment]
You can add at source code.

**Comment by viragumathe5** [comment]
Like by just commenting on it ???

**Comment by viragumathe5** [comment]
Ohhhhh I am sorry so API documentation is for the auto code generation. I  am sorry I just misunderstood the documentation. I am on ok

**Comment by viragumathe5** [comment]
Ohhhhh I am sorry so API documentation is for the auto doc generation. I am sorry I just misunderstood the documentation. I am on ok
I am proceeding with docstrings as mentioned.

Thank You for clearing that....

**Comment by kmehant** [comment]
@dexhunter Can I work on converting the docstrings from google style to sphinx style.

**Comment by dexhunter** [comment]
@kmehant  Sure, feel free to pick up the issue

**Comment by dexhunter** [comment]
fixed at #94
